### ISSUE TITLE
WCAG 2.0 requires role="listbox" to be labelled

### ISSUE THREAD

**Issue by Htbaa** [WebDevStudios/wp-search-with-algolia#65]
**Describe the bug**
According to WCAG 2.0 the use of `role="listbox"` requires the element to be described by a label with either `aria-label` or `aria-labbeledby` (see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role as well).

The Javascript generates the following HTML which doesn't meet these requirements:

    <span class="algolia-autocomplete" style="position: absolute; z-index: 100; display: none; direction: ltr;">
        <span class="aa-dropdown-menu" role="listbox" id="algolia-autocomplete-listbox-0" style="display: block; left: 0px; right: auto;">
            <div class="aa-empty" style="display: none;"></div>
            <div class="aa-dataset-1" style></div>
        </span>
    </span>
    <span class="algolia-autocomplete" style="position: absolute; z-index: 100; display: none; direction: ltr;">
        <span class="aa-dropdown-menu" role="listbox" id="algolia-autocomplete-listbox-2" style="display: block; left: 0px; right: auto;">
            <div class="aa-empty" style="display: none;"></div>
            <div class="aa-dataset-3" style></div>
        </span>
    </span>

**Expected behavior**
Elements that have `role="listbox"` should be labelled to meet the WCAG 2.0 requirements.

**Additional context**
This issue popped up using a website analyzer for accessibility.

**Comment by richaber** [comment]
Howdy @Htbaa,

We use [autocomplete.js](https://github.com/algolia/autocomplete.js/) from Algolia for the autocomplete dropdown functionality.

We currently use version 0.28.3, and I do not see an option to pass along either an aria-label or aria-labbeldby to the listbox markup ([reference here](https://github.com/algolia/autocomplete.js/blob/v0.28.3/dist/autocomplete.jquery.js#L882-L890)).

That particular portion of the code doesn't appear to have changed much in the latest version either ([reference here](https://github.com/algolia/autocomplete.js/blob/v0.37.1/dist/autocomplete.jquery.js#L903-L911)).

This might be an issue to bring to their attention?

**Comment by Htbaa** [comment]
Hi @richaber,

I had no idea that part of the project was still in the hands of Algolia. If it's them maintaining the JavaScript then I'll forward the issue to them. Let's keep this issue open for the time being until the issue has been resolved on their end?

PS: is there a reason for being behind so many releases? It seems that 0.37.1 was released January 27th of this year.

**Comment by richaber** [comment]
Howdy @Htbaa 

When we forked from the (abandoned) official plugin, we inherited the Algolia PHP and JS library versions that had last shipped with that plugin.

The main reason the JS libs are not up to date with their latest release versions, are to avoid "breaking changes."

That said, the latest [WP Search with Algolia](https://wordpress.org/plugins/wp-search-with-algolia/) release, [1.4.0](https://github.com/WebDevStudios/wp-search-with-algolia/releases/tag/1.4.0) updates the Algolia PHP client to the latest release version, and updates the Algolia JS libraries to their most recent _non-breaking_ versions. That includes autocomplete.js version [0.37.1](https://github.com/algolia/autocomplete.js/releases/tag/v0.37.1), which I believe is the one you are most interested in at the moment.

**Comment by Htbaa** [comment]
Hi @richaber,

That's great to hear! Will schedule testing this version. Until now I've been using some additional Javascript to add the stuff that was missing. Would be nice if that patchwork will no longer be necessary.
